java-symbol-solver-logic/src/main/java/com/github/javaparser/symbolsolver/logic/InferenceContext.java

```diff
@@ -75,7 +75,14 @@ private void registerCorrespondance(Type formalType, Type actualType) {
                 final String formalParamTypeQName = formalTypeAsReference.getQualifiedName();
                 List<Type> correspondingFormalType = ancestors.stream().filter((a) -> a.getQualifiedName().equals(formalParamTypeQName)).collect(Collectors.toList());
                 if (correspondingFormalType.isEmpty()) {
-                    throw new ConfilictingGenericTypesException(formalType, actualType);
+                    List<ReferenceType> ancestors2 = formalTypeAsReference.getAllAncestors();
```

Comment:
I'll rename it to reuse `ancestor` from above. I'll try to refactor that method at some point because sometimes arguments come in the opposite order `formal/actual` and there is some duplicated code to support the other way around